### PR TITLE
dist: add script for generating bundle with dependencies

### LINKED ISSUE #442: Unified build and deployment to CDN
Sanctuary currently requires installing different modules (sanctuary-def, sanctuary-type-classes, etc.)

This is handled elegantly and automatically by package managers such as npm, however there are several use cases where loading via CDN would be preferred - and in those cases each module must be imported separately which can cause a bit of friction.

Suggestion is to create a concatenation+uglify build tool and push to a popular cdn host like unpkg or cdnjs

@JAForbes suggested on Gitter that deployment to unpkg is relatively simple and automated once you setup the initial script.

This is also currently being done by Ramda, fwiw:

https://unpkg.com/ramda@0.24.1/dist/ramda.min.js / https://cdnjs.com/libraries/ramda

### LINKED ISSUE #480: submit to CDNJS
I just found myself wanting to fetch Sanctuary from [CDNJS][1]. Let's make this possible. :)

The process of adding a library to CDNJS is daunting to the uninitiated. If you have experience with the process and would like to be involved, please speak up. :raising_hand_woman:

[1]: https://cdnjs.com/

### LINKED ISSUE #613: provide a min version for browser
it's better to build a minimize version for browser, thank you!

### PR DESCRIPTION
**Pull request by davidchambers**
Closes #613; closes #480; closes #443; closes #442

Should the files be included in the GitHub repository? The npm package? Both?

|                           | GitHub repository | npm package       |
| ------------------------- |:-----------------:|:-----------------:|
| __dist/bundle.js__        | :question:        | :question:        |
| __dist/bundle.min.js__    | :question:        | :question:        |

/cc @loveship00, @mfidemraizer

### CODE REVIEW

**Comment by Avaq** [comment]
I would say that these files should be included in the GitHub repository for automatic distribution through platforms such as [JSDelivr](https://www.jsdelivr.com/). They should be excluded from the npm package, as npm is a code delivery platform that specifically targets environments capable of managing their own dependencies.

**Comment by Avaq** [comment]
Note that JSDelivr offers minification through its own server. For example, this is the minified version of Fluture as generated by JSDelivr: https://cdn.jsdelivr.net/gh/fluture-js/Fluture@12.0.0-beta.2/dist/bundle.min.js

It's therefore not strictly necessary to include a minified file alongside the regular file.

**Comment by davidchambers** [comment]
> Note that JSDelivr offers minification through its own server.

Cool! I'll remove the UglifyJS dependency.

**Comment by Avaq** [comment]
Doesn't this also close the following PR's and issues?

- #480
- #443
- #442

File: scripts/prepublish [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+set -euf -o pipefail
+
+node_modules/.bin/sanctuary-prepublish "$@"
+
+rm -rf node_modules
+npm install
+mkdir -p -- dist
+scripts/bundle >dist/bundle.js
```
Comment (davidchambers) [inline scripts/prepublish]: I updated `scripts/bundle` to write to standard output, to make it apparent that `mkdir -p` must be used.

**Pull request merged**